aws-lc-rs/tests/pqdsa_test.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+#![cfg(all(not(feature = "fips"), feature = "unstable"))]
+
+use aws_lc_rs::signature::{KeyPair, VerificationAlgorithm};
+use aws_lc_rs::unstable::signature::{
+    PqdsaKeyPair, MLDSA_44, MLDSA_44_SIGNING, MLDSA_65, MLDSA_65_SIGNING, MLDSA_87,
+    MLDSA_87_SIGNING,
+};
+use aws_lc_rs::{test, test_file};
+
+macro_rules! mldsa_keygen_test {
+    ($file:literal, $signing:expr) => {
+        test::run(test_file!($file), |section, test_case| {
+            assert_eq!(section, "");
+            let _seed = test_case.consume_bytes("SEED");
+            let public = test_case.consume_bytes("PUBLIC");
+            let secret = test_case.consume_bytes("SECRET");
+
+            let key_pair_secret = PqdsaKeyPair::from_raw_private_key($signing, secret.as_slice())?;
+            let public_secret = key_pair_secret.public_key();
+            assert_eq!(public.as_slice(), public_secret.as_ref());
+
+            Ok(())
+        });
+    };
+}
+
+macro_rules! mldsa_sigver_test {
+    ($file:literal, $verification:expr) => {
+        test::run(test_file!($file), |section, test_case| {
+            assert_eq!(section, "");
+            let public_key = test_case.consume_bytes("PUBLIC");
+            let message = test_case.consume_bytes("MESSAGE");
+            let signature = test_case.consume_bytes("SIGNATURE");
+            let context = test_case.consume_bytes("CONTEXT");
+            let expected_result = test_case.consume_bool("RESULT");
+
+            if !context.is_empty() {
+                // skip test
+                return Ok(());
+            }
```

Comment:
What's the reason for skipping tests without empty `context`s? Would we unintentionally skipping test cases if we imported files with populated `context`s?

Reply:
Many of the ML-DSA KATs require a specific context to be set. This is useful for internally testing the AWS-LC implementation. However, AWS-LC doesn't (yet) expose the APIs necessary for consumers (like `aws-lc-rs`) to specify a context and then match our answer. So in `aws-lc-rs`, we can only compare our results against KATs where the context is empty.

Originally I kept all of the KATs in our data files, but I've since removed then. I forgot to also remove this condition from the test.